packages/adapters/ethers/src/client.ts

```diff
@@ -613,12 +613,10 @@ export class EthersAdapter {
       if (provider) {
         const { addresses, chainId } = await EthersHelpersUtil.getUserInfo(provider)
         const firstAddress = addresses?.[0]
-        const caipNetwork = this.caipNetworks.find(c => c.chainId === chainId)
         const caipAddress = `${this.chainNamespace}:${chainId}:${firstAddress}` as CaipAddress
 
-        if (firstAddress && chainId && caipNetwork) {
+        if (firstAddress && chainId) {
           this.appKit?.setCaipAddress(caipAddress, this.chainNamespace)
-          this.appKit?.setCaipNetwork(caipNetwork)
```

Comment:
We should do all the `caipNetwork` setters from a single place and do it only when it's necessary.

Reply:
++++++